packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1489 @@
+    // be automatically added for them when generating descriptions of the
+    // property.
+    assert(name == null || !name.endsWith(':'));
+  }
+
+  /// Constructor that creates a [DiagnosticsNode] where properties and children
+  /// are computed lazily.
+  factory DiagnosticsNode.lazy({
+    String name,
+    T object,
+    String description,
+    void fillProperties(List<DiagnosticsNode<Object>> properties),
+    List<DiagnosticsNode<Object>> getChildren(),
+    String emptyBodyDescription,
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.sparse,
+  }) {
+    return new _LazyMembersDiagnosticsNode<T>(
+      name: name,
+      object: object,
+      description: description,
+      fillProperties: fillProperties,
+      getChildren: getChildren,
+      style: style,
+      emptyBodyDescription: emptyBodyDescription,
+    );
+  }
+
+
+  /// Diagnostics containing just a string `message` and not a a concrete name
+  /// or value.
+  static DiagnosticsNode<Object> message(String message, {
+    DiagnosticsTreeStyle style: DiagnosticsTreeStyle.singleLine,
+  }) {
+    return new DiagnosticsProperty<Object>(
+      '',
+      null,
+      description: message,
+      style: style,
+    );
+  }
+
+  /// Label describing the Diagnostics node.
+  final String name;
+
+  /// Description with a short summary of the node itself not including children
+  /// or properties.
+  String get description;
+
+  /// Whether to show a separator between [name] and [description].
+  ///
+  /// If false, name and description should be shown with no separation.
+  /// `:` is typically used as a separator when displaying as text.
+  final bool showSeparator;
+
+  /// Whether the diagnostics should be hidden when showing the default
+  /// view of a tree.
+  bool get hidden;
+
+  final bool showName;
+
+  /// Description to show if the node has no displayed properties or children.
+  final String emptyBodyDescription;
+
+  /// Dart object this is diagnostics data for.
+  T get object;
+
+  /// Hint for how the node should be displayed.
+  final DiagnosticsTreeStyle style;
+
+  /// Properties of this DiagnosticsNode.
+  ///
+  /// Properties and children are kept distinct even though they are both
+  /// [List<DiagnosticNode>] because they should be grouped differently.
+  List<DiagnosticsNode<Object>> getProperties();
+
+  /// Children of this DiagnosticsNode.
+  ///
+  /// See also:
+  ///
+  ///  * [getProperties]
+  List<DiagnosticsNode<Object>> getChildren();
+
+  String get _separator => showSeparator ? ':' : '';
+
+  @override
+  String toString() {
+    if (style == DiagnosticsTreeStyle.singleLine)
+      return toStringDeep();
+
+    if (name == null || name.isEmpty || showName == false)
+      return description;
+
+    return description.contains('\n') ?
+        '$name$_separator\n$description' : '$name$_separator $description';
+  }
+
+  TextTreeConfiguration get textTreeConfiguration {
+    switch (style) {
```

Comment:
TextTreeConfiguration is specific to displaying DiagnosticsNode objects as text.
DiagnosticsTreeStyle applies equally in a text view of a DiagnosticsNode object and an IntelliJ tree view displaying styled DiagnosticsNode objects. All of the TextTreeConfiguration is meaningless in IntelliJ as colors and/or icons will be used instead of ascii art to distinguish the styles.

`get textTreeConfiguration`
should really be protected as it only needs to be used internally by `toStringDeep`.